### PR TITLE
EIP 1559 introductory support

### PR DESCRIPTION
**Pull request by marcgarreau**
### What was wrong?

Etheriom keeps updating.

(Related to #1835)

### How was it fixed?

This is early, exploratory code:
- wrote some test cases (probably not comprehensive)
- parsed out legacy vs. 1559 txs in the `gas_price_strategy_middleware`
- ~pulled latest Geth from master & built from source~ London now available in Geth v1.10.4
- generated a London test fixture from Geth v1.10.4 (`common_1559.py` is temporary for this purpose)
- ran the integration test suite through the new fixture to see what explodes

### Todo:
- [x] input munging
- [ ] revisit gas price generation
- [ ] debug a couple failing tests
- [x] update docs for sending txs (`maxFeePerGas` & `maxPriorityFeePerGas`)
- [ ] update docs for retrieving blocks (`baseFeePerGas`)
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

### CODE REVIEW

**Comment by marcgarreau** [comment]
Theoretically [London goes live](https://blog.ethereum.org/2021/06/18/london-testnets-announcement/) on Ropsten tomorrow, so its about time to get this work off my fork and into a Web3.py branch for easier use on testnets. Once this is up on a branch, I can remove myself as the gatekeeper and anyone can create PRs against it. I've created a `london` branch, and this PR now points at it.

Current state: 
- Core functionality is there and ready for testnet testing. There's an edge case or two that needs a design decision or polish.
- There are a couple of grumpy tests that we still need to track down; I don't believe their failures indicate anything mission critical, but rather some fine tuning that needs to happen/cleanup where some tests are polluting the context of those failing tests.
- One of the bigger TODOs: gas price generation code needs a hard look. A naive version may be the short term solution, which 'smarter' ones are figured out, e.g., something involving [`feeHistory`](https://github.com/ethereum/go-ethereum/pull/23033).

File: web3/_utils/method_formatters.py [hunk 1]
```diff
@@ -145,13 +145,15 @@ def is_attrdict(val: Any) -> bool:
 not_attrdict = complement(is_attrdict)
 
 
-TRANSACTION_FORMATTERS = {
+TRANSACTION_RESULT_FORMATTERS = {
```
Comment (kclowes) [inline web3/_utils/method_formatters.py:148]: 👍  nice change!

File: web3/middleware/gas_price_strategy.py [hunk 2]
```diff
@@ -27,11 +32,44 @@ def gas_price_strategy_middleware(
...
+                    #  raise InvalidTransaction('maxFeePerGas must be >= maxPriorityFeePerGas')
+                transaction = assoc(transaction, 'maxFeePerGas', hex(base_fee * 2))
+                return make_request(method, [transaction])
+            # 1559 - no tip:
+            elif 'maxFeePerGas' in transaction and 'maxPriorityFeePerGas' not in transaction:
+                raise InvalidTransaction(
+                    "maxPriorityFeePerGas must be defined in a 1559 transaction."
+                )
+            # legacy tx:
+            elif 'gasPrice' not in transaction:
```
Comment (kclowes) [inline web3/middleware/gas_price_strategy.py:64]: Should this be put back before merging? I think this changes the default behavior if I'm reading it right?

**Review by kclowes — APPROVED** [review]
LGTM for branch merge! Just left one question 🚀

File: web3/middleware/gas_price_strategy.py (hunk 2 above)
Comment (marcgarreau) [reply web3/middleware/gas_price_strategy.py:64]: Yeah, I think that's right. This is a noop anyway, so will just nix this condition now.

**Review by fselmo — APPROVED** [review]
